Index: miniunz.c
===================================================================
--- miniunz.c	(revision 3060)
+++ miniunz.c	(working copy)
@@ -519,7 +519,7 @@
 }
 
 
-int main(argc,argv)
+int miniunz(argc,argv)
     int argc;
     char *argv[];
 {
@@ -633,7 +633,7 @@
 #endif
         {
           printf("Error changing into %s, aborting\n", dirname);
-          exit(-1);
+          return -1; /* exit(-1); */
         }
 
         if (filename_to_extract == NULL)
Index: miniunz.h
===================================================================
--- miniunz.h	(revision 0)
+++ miniunz.h	(working copy)
@@ -0,0 +1,6 @@
+#ifndef MINIUNZ_H
+#define MINIUNZ_H
+
+int miniunz(int argc , char *argv[]); /* Renamed the main function */
+
+#endif /* End of header file MINIUNZ_H */
\ No newline at end of file
