Index: kinsol.c
===================================================================
--- kinsol.c	(revision 6701)
+++ kinsol.c	(working copy)
@@ -1403,8 +1403,7 @@
           rldiff = rldiff - rlinc;
         }
 
-      } while ((*f1normp > alpha_cond) ||
-	       ((*f1normp < beta_cond) && (rldiff >= rlmin)));
+      } while (((*f1normp > alpha_cond) || (*f1normp < beta_cond)) && (rldiff >= rlmin));
 
       if ((*f1normp) < beta_cond) {
 
