Index: kinsol.c
===================================================================
--- kinsol.c	(revision 7140)
+++ kinsol.c	(working copy)
@@ -137,7 +137,7 @@
  * MAX_RECVR   max. no. of attempts to correct a recoverable func error
  */
 
-#define MAX_RECVR      5
+#define MAX_RECVR      20
 
 /*
  * Keys for KINPrintInfo
--- kinsol.c	(revision 6701)
+++ kinsol.c	(working copy)
@@ -1403,8 +1403,7 @@
           rldiff = rldiff - rlinc;
         }
 
-      } while ((*f1normp > alpha_cond) ||
-	       ((*f1normp < beta_cond) && (rldiff >= rlmin)));
+      } while (((*f1normp > alpha_cond) || (*f1normp < beta_cond)) && (rldiff >= rlmin));
 
       if ((*f1normp) < beta_cond) {
 
