Index: sundials-2.5.0/src/kinsol/kinsol.c
===================================================================
--- sundials-2.5.0/src/kinsol/kinsol.c	(revision 8553)
+++ sundials-2.5.0/src/kinsol/kinsol.c	(working copy)
@@ -1405,9 +1405,12 @@
 
       } while (((*f1normp > alpha_cond) || (*f1normp < beta_cond)) && (rldiff >= rlmin));
 
-      if ((*f1normp) < beta_cond) {
 
-	/* beta condition could not be satisfied so set unew to last u value
+
+      if ((*f1normp) < beta_cond || (rldiff < rlmin && (*f1normp > alpha_cond))) {
+
+	/* beta condition could not be satisfied or rldiff to small and alpha_cond not satisfied
+	   so set unew to last u value
 	   that satisfied the alpha condition and continue */
 
         N_VLinearSum(ONE, uu, rllo, pp, unew);
